src/python/library/tritonclient/grpc/aio/__init__.py

```diff
@@ -586,8 +586,35 @@ async def infer(
         headers=None,
         compression_algorithm=None,
         parameters=None,
+        get_call_obj=False,
     ):
-        """Refer to tritonclient.grpc.InferenceServerClient"""
+        """Refer to tritonclient.grpc.InferenceServerClient
+        The additional parameters for this functions are
+        described below:
+
+        Parameters
+        ----------
+        get_call_obj : bool
+            If set True, then this function will yield
+            grpc.aio.call object first bfore the
+            InferResult.
+            This object can be used to issue request
+            cancellation if required. This can be attained
+            by following:
+            -------
+            generator = client.infer(..., get_call_obj=True)
+            grpc_call = await anext(generator)
+            grpc_call.cancel()
+            -------
```

Comment:
Would it be possible to wrap the call object in another class (perhaps `Request`) that once awaited would provide and has a cancel method?

Reply:
There are some new primitives that got added for async_generators in python 3.10. I am not very familiar with these right out of bat first trial was not successful. 
Reverting cancellation support in asyncio for now. Will be addressing the issue later.